service_implementation/advanced_plm.md

```diff
@@ -1 +1,456 @@
-# Beyond just files: Using PLM for versioning and metadata
\ No newline at end of file
+# Beyond just files: Using PLM for versioning and metadata
+This document describes how to use PLM services from _inside_ a service, i.e., by calling the PLM API. The text includes  example code.
+There are two types of PLM services for use in CAxMan; both are described below:
+1) those that are integrated in the GSS framework;
+2) the PLM Server native web-services.
+
+## PLM API through GSS
+
+The PLM Server has its own implementation of the GSS services, that is, PLM service endpoints can be obtained through GSS. GSS offers SOAP and REST services:
+- WSDL (SOAP): https://caxman.clesgo.net/jotne/GSS/FileWebService?wsdl
+- REST end point: https://caxman.clesgo.net/jotne/GSS/resources/plm
+
+GSS covers a small subset of simplified methods (services) of the PLM Server. The remaining PLM Server functionality is only available as native PLM Server SOAP web-services; see below for details.
+
+Java (Maven) client based examples can be retrieved from here: https://github.com/CAxMan/PlmGSS-samples
+
+The PLM Server is also accessible via the GSS.
+Code samples of that are here: https://github.com/CAxMan/infrastructureClients
+
+The following GSS methods (SOAP) are only implemented for the PLM Server storage option, not for the SWIFT storage:
+- *createFolderVersion*
+- *listFolderVersions*
+- *listFileVersions*
+- *plm2gss*
+- *gss2plm*
+
+**createFolderVersion** - creates a new version of the specified folder. Children folders of the existing selected version are reused; files are copied.
+Parameters:
+- **folderID** - GSS folder id of the folder to create the new version from. Example: plm://InitialRepository/Ultralight_Glider/236223201541
+- **session_id** - Keystone authentication token
+Returns: **ResourceInformation** object with GSS folder ID of the newly created folder version
+
+**listFolderVersions** - lists all versions of the specified folder including the one specified.
+Parameters:
+- **path** - GSS folder id. Example: plm://InitialRepository/Ultralight_Glider/236223201541
+- **session_id** - Keystone authentication token
+Returns: **ResourceInformation** object for each folder version
+
+**listFileVersions** - lists all versions of the specified file including the one specified.
+Parameters:
+- **path** - GSS file id. Example: plm://InitialRepository/Ultralight_Glider/236223201541
+- **session_id** - Keystone authentication token
+Returns: **ResourceInformation** object for each file version
+

```

Comment:
Suggested addition (links nicely with the paragraph below):
Note that the above methods can technically accept any kind of GSS URI (which is represented as a mere string in CAxMan), but the methods will obviously will when non-PLM URIs are provided.

Reply:
I suggest to simply add the text from the above comment:

"Note that the above methods can technically accept any kind of GSS URI (which is represented as a mere string in CAxMan), but the methods will obviously fail when non-PLM URIs are provided."